### PR TITLE
Polimorfic association for transactions

### PR DESCRIPTION
**Pull request by marderer**
- Add fields to transactions for polymorphic and delete unnecessary fields.
- Create a model for a transaction to adding money to the balance.
- Create a model for a transaction for the distribution of money by category.
- Create a model for a transaction for  the distribution of money between categories.
- Add tasks for converting transactions to BalanceTransaction. Just run `rails user:convert_transactions`.
- Add tests for all created models
- Сhange in the calculation of the balance.

### CODE REVIEW

File: app/models/category.rb [hunk 1]
```diff
@@ -1,5 +1,8 @@
 class Category < ApplicationRecord
   belongs_to :user
+  has_many :category_transactions
+  has_many :category_from, class_name: 'BetweenCategoriesTransaction', foreign_key: 'category_from_id'
+  has_many :category_to, class_name: 'BetweenCategoriesTransaction', foreign_key: 'category_to_id'
```
Comment (marderer) [inline app/models/category.rb]: I don't know whether it is correct. Maybe I changing it late.

File: .rubocop.yml [hunk 2]
```diff
@@ -199,6 +199,15 @@ Style/StringLiterals:
...
   Enabled: true
 
+Rails/CreateTableWithTimestamps:
+  Enabled: false
+
+Rails/HasManyOrHasOneDependent:
+  Enabled: false
+
+Rails/InverseOf:
+  Enabled: false
```
Comment (AleksSenkou) [inline .rubocop.yml]: @marderer why did you disable all those metrics?

Comment (AleksSenkou) [reply .rubocop.yml]: Having timestamps everywhere is a good practice 😉

Comment (marderer) [reply .rubocop.yml]: @AleksSenkou In this article https://launchschool.com/blog/understanding-polymorphic-associations-in-rails, wrote that `InverseOf` does not work with polymorfic, but rubocop required add it. 
About timestamps I asked in slack.

Comment (marderer) [reply .rubocop.yml]: @AleksSenkou I added timestamps and deleted cop.

Comment (AleksSenkou) [reply .rubocop.yml]: @marderer thank you, got it

**Pull request merged**